packages/device_info/device_info_platform_interface/lib/method_channel_device_info.dart

```diff
@@ -9,14 +11,20 @@ const MethodChannel _channel = MethodChannel('plugins.flutter.io/device_info');
 /// An implementation of [DeviceInfoPlatform] that uses method channels.
 class MethodChannelDeviceInfo extends DeviceInfoPlatform {
   // Method channel for Android devices
-  Future<Map<String, dynamic>> androidInfo() async {
-    return Map<String, dynamic>.from(
-        await _channel.invokeMethod('getAndroidDeviceInfo'));
+  Future<AndroidDeviceInfo> androidInfo() async {
+    return AndroidDeviceInfo.fromMap(
+      Map<String, dynamic>.from(
```

Comment:
Do you need to wrap the result of _channel.invokeMethod in the Map constructor in this case?

Reply:
I thought it would be safer this way because when I was testing it on linux it gave me this error : `'_InternalLinkedHashMap<dynamic, dynamic>' is not a subtype of type
'Map<String, dynamic>' in type cast` so to keep it safe for all, I applied this type cast.

Reply:
Ah, you can also use `.cast<String, dynamic>()` on the map returned from the invokeMethod so you don't have to create a clone of the result:

See this one: https://stackoverflow.com/a/51283565

Reply:
You mean like this `(await _channel.invokeMethod('getIosDeviceInfo')).cast<String, dynamic>()`?

Reply:
Yep